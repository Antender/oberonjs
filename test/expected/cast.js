var RTL$ = {
	extend: function extend(methods){
		methods.__proto__ = this.prototype; // make instanceof work

		// to see constructor name in diagnostic
		var result = methods.init;
		methods.constructor = result.prototype.constructor;

		result.prototype = methods;
		result.extend = extend;
		return result;
	},
	typeGuard: function (from, to){
		if (!(from instanceof to))
			throw new Error("typeguard assertion failed");
		return from;
	}
};
var m = function (){
var Base = RTL$.extend({
	init: function Base(){
	}
});
var Derived1 = Base.extend({
	init: function Derived1(){
		Base.prototype.init.bind(this)();
		this.field1 = 0;
	}
});
var Derived2 = Derived1.extend({
	init: function Derived2(){
		Derived1.prototype.init.bind(this)();
		this.field2 = 0;
	}
});
var pb = null;
var pd1 = null;
var pd2 = null;
pd2 = new Derived2();
pb = pd2;
pd1 = pd2;
RTL$.typeGuard(pb, Derived1).field1 = 0;
RTL$.typeGuard(pb, Derived2).field2 = 1;
RTL$.typeGuard(pd1, Derived2).field2 = 2;
}();