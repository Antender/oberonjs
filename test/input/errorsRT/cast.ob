MODULE m1;
TYPE
    T* = RECORD
    END;

END m1.

MODULE m2;
IMPORT m1;
TYPE
    T = RECORD(m1.T)
    END;
    PT = POINTER TO T;
VAR
    p: POINTER TO m1.T;
BEGIN
    NEW(p);
    ASSERT(p(PT) # NIL);
END m2.