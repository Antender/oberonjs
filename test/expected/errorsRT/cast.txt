Error: cannot cast to 'm2.T' from 'm1.T'